compliance/collection/nodeinventorizer/caching_inventorizer_test.go

```diff
@@ -0,0 +1,251 @@
+	"github.com/stackrox/rox/pkg/jsonutil"
+	"github.com/stretchr/testify/suite"
+)
+
+type mockSleeper struct {
+	receivedDuration time.Duration
+	callCount        int
+}
+
+func (ms *mockSleeper) mockWaitCallback(d time.Duration) {
+	ms.receivedDuration = d
+	ms.callCount++
+}
+
+type mockScanner struct{}
+
+func (m mockScanner) Scan(nodeName string) (*storage.NodeInventory, error) {
+	return &storage.NodeInventory{NodeName: nodeName}, nil
+}
+
+type TestComplianceCachingSuite struct {
+	suite.Suite
+}
+
+func TestComplianceCaching(t *testing.T) {
+	suite.Run(t, new(TestComplianceCachingSuite))
+}
+
+func (s *TestComplianceCachingSuite) writeWrap(wrap *inventoryWrap, path string) {
+	jsonWrap, err := json.Marshal(&wrap)
+	s.NoError(err)
+	err = os.WriteFile(path, jsonWrap, 0600)
+	s.NoError(err)
+}
+
+func (s *TestComplianceCachingSuite) readWrapToInventory(path string) *storage.NodeInventory {
+	cacheContents, err := os.ReadFile(path)
+	s.NoError(err)
+
+	var wrap inventoryWrap
+	err = json.Unmarshal(cacheContents, &wrap)
+	s.NoError(err)
+
+	var testInv storage.NodeInventory
+	err = jsonutil.JSONToProto(wrap.CachedInventory, &testInv)
+	s.NoError(err)
+
+	return &testInv
+}
+
+func (s *TestComplianceCachingSuite) TestMin() {
+	cases := map[string]struct {
+		a    time.Duration
+		b    time.Duration
+		want time.Duration
+	}{
+		"a smaller": {
+			a:    time.Second,
+			b:    time.Hour,
+			want: time.Second,
+		},
+		"b smaller": {
+			a:    time.Hour,
+			b:    time.Second,
+			want: time.Second,
+		},
+		"a zero": {
+			a:    0 * time.Second,
+			b:    time.Second,
+			want: 0,
+		},
+		"same unit - a smaller": {
+			a:    2 * time.Second,
+			b:    10 * time.Second,
+			want: 2 * time.Second,
+		},
+		"same unit - b smaller": {
+			a:    10 * time.Second,
+			b:    2 * time.Second,
+			want: 2 * time.Second,
+		},
+		"different unit - a smaller": {
+			a:    2 * time.Second,
+			b:    7 * time.Hour,
+			want: 2 * time.Second,
+		},
+		"different unit - b smaller": {
+			a:    2 * time.Second,
+			b:    100 * time.Millisecond,
+			want: 100 * time.Millisecond,
+		},
+	}
+	for k, v := range cases {
+		s.Run(k, func() {
+			actual := min(v.a, v.b)
+			s.Equal(v.want, actual)
+		})
+	}
+}
+
+func (s *TestComplianceCachingSuite) TestCalcNextBackoffTable() {
+	cases := map[string]struct {
+		initial         time.Duration
+		maxBackoff      time.Duration
+		expectedBackoff time.Duration
+	}{
+		"next backoff should not hit the limit of 30s": {
+			initial:         10 * time.Second,
+			maxBackoff:      30 * time.Second,
+			expectedBackoff: 15 * time.Second,
+		},
+		"next backoff should not be higher than the limit of 10s": {
+			initial:         8 * time.Second,
+			maxBackoff:      10 * time.Second,
+			expectedBackoff: 10 * time.Second,
+		},
+	}
+	for name, c := range cases {
+		s.Run(name, func() {
+			cs := NewCachingScanner(mockScanner{}, "", c.initial, c.initial, c.maxBackoff, func(time.Duration) {})
```

Comment:
[nit] The third parameter to the call, `c.initial` has a meaning of `cacheDuration`. It confused me first that `c.initial` is passed two times. I suggest passing a different value to `cacheDuration`. E.g.
```suggestion
			cs := NewCachingScanner(mockScanner{}, "", 1*time.Hour, c.initial, c.maxBackoff, func(time.Duration) {})
```